vsphere/datadog_checks/vsphere/utils.py

```diff
@@ -51,40 +58,66 @@ def match_any_regex(string, regexes):
     return False
 
 
-def is_resource_excluded_by_filters(mor, infrastructure_data, resource_filters):
-    # type: (vim.ManagedEntity, InfrastructureData, ResourceFilters) -> bool
+def is_resource_collected_by_filters(mor, infrastructure_data, resource_filters, tags_cache):
+    # type: (vim.ManagedEntity, InfrastructureData, ResourceFilters, TagsCache) -> bool
     resource_type = MOR_TYPE_AS_STRING[type(mor)]
 
-    if not [f for f in resource_filters if f[0] == resource_type]:
-        # No filter for this resource, collect everything
+    if not [f for f in resource_filters if f.resource == resource_type and f.is_whitelist]:
+        # No whitelist filter specified for this resource, consider that the resource matches the whitelist
+        match_whitelist = True
+    else:
+        match_whitelist = _does_resource_match_filters(
+            mor, infrastructure_data, resource_filters, tags_cache, is_whitelist=True
+        )
+
+    match_blacklist = _does_resource_match_filters(
+        mor, infrastructure_data, resource_filters, tags_cache, is_whitelist=False
+    )
+
+    if match_blacklist:
+        # If resources matches one of the blacklisted patterns, do not collect it
         return False
 
-    name_filter = resource_filters.get((resource_type, 'name'))
-    inventory_path_filter = resource_filters.get((resource_type, 'inventory_path'))
-    hostname_filter = resource_filters.get((resource_type, 'hostname'))
-    guest_hostname_filter = resource_filters.get((resource_type, 'guest_hostname'))
+    # Otherwise, collect it if it matches one of the whitelisted patterns.
+    return match_whitelist
+
+
+def _does_resource_match_filters(mor, infrastructure_data, resource_filters, tags_cache, is_whitelist=True):
```

Comment:
Looks like the resource filter concept/logic is not trivial anymore.

We should probably create separate classes to deal with that. For example a `ResourceFilterConfig` class, that contain a list/dict of individual `ResourceFilter`(s).

That way we can encapsulate the logic in those classes and avoid having logic in utils.py.

With a public method like `ResourceFilterConfig.is_collectted()` and encapsulate the rest.